master/internal/db/postgres_rp_workspace_bindings.go

```diff
@@ -170,3 +168,91 @@ func OverwriteRPWorkspaceBindings(ctx context.Context,
 	err = AddRPWorkspaceBindings(ctx, workspaceIds, poolName, resourcePools)
 	return err
 }
+
+// GetUnboundRPs get unbound resource pools.
+func GetUnboundRPs(
+	ctx context.Context, resourcePools []config.ResourcePoolConfig,
+) ([]string, error) {
+	var boundResourcePools []string
+	_, err := Bun().NewSelect().
+		Column("pool_name").
+		Table("rp_workspace_bindings").
+		Distinct().
+		Exec(ctx, &boundResourcePools)
+	if err != nil {
+		return nil, err
+	}
+
+	boundRPsMap := map[string]bool{}
+	for _, boundRP := range boundResourcePools {
+		boundRPsMap[boundRP] = true
+	}
+
+	unboundRPs := []string{}
+	for _, resourcePool := range resourcePools {
+		if !boundRPsMap[resourcePool.PoolName] {
+			unboundRPs = append(unboundRPs, resourcePool.PoolName)
+		}
+	}
+
+	return unboundRPs, nil
+}
+
+// RP is a helper strct for Bun query.
+type RP struct {
+	Name string
+}
+
+// ReadRPsAvailableToWorkspace returns the names of resource pool bound to a
+// workspace.
+func ReadRPsAvailableToWorkspace(
+	ctx context.Context,
+	workspaceID int32,
+	offset int32,
+	limit int32,
+	resourcePoolConfig []config.ResourcePoolConfig,
+) ([]string, *apiv1.Pagination, error) {
+	unboundRPNames, err := GetUnboundRPs(ctx, resourcePoolConfig)
+	if err != nil {
+		return nil, nil, err
+	}
+	unboundRPs := []*RP{}
+	for _, unboundRPName := range unboundRPNames {
+		unboundRPs = append(unboundRPs, &RP{unboundRPName})
+	}
+
+	var rpNames []string
+	var query *bun.SelectQuery
+	if len(unboundRPs) > 0 {
+		values := Bun().NewValues(&unboundRPs)
+		boundAndUnboundRPSubTable := Bun().NewSelect().
+			ColumnExpr("pool_name AS Name").
+			Table("rp_workspace_bindings").
+			Where("workspace_id = ?", workspaceID).
+			UnionAll(Bun().NewSelect().With("unboundRP", values).Table("unboundRP"))
+		query = Bun().NewSelect().
+			TableExpr("(?) AS rp", boundAndUnboundRPSubTable)
+	} else {
+		query = Bun().NewSelect().
+			ColumnExpr("pool_name AS Name").
+			Table("rp_workspace_bindings").
+			Where("workspace_id = ?", workspaceID)
+	}
+
+	pagination, query, err := getPagedBunQuery(ctx, query, int(offset), int(limit))
+	if err != nil {
+		return nil, nil, err
+	}
+	// Bun bug treating limit=0 as no limit when it
+	// should be the exact opposite of no records returned.
+	if pagination.StartIndex-pagination.EndIndex != 0 {
```

Comment:
I don't think this is a bug - if limit is 0, we'll pass thru all the information

Reply:
I think this is for the case limit = -2 returns pagination info but no experiments. In this case StartIndex = EndIndex.

Reply:
maybe make a TODO comment and we can revisit later, especially if we're trying to reuse this getPagedBunQuery function

Reply:
e.g. `// TODO: revisit and check this for commonality`